src/Compilers/CSharp/Portable/Symbols/Extensions/RewrittenExtensionMethodSymbol.cs

```diff
@@ -0,0 +1,51 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics;
+using Microsoft.CodeAnalysis.CSharp.Emit;
+using Microsoft.CodeAnalysis.PooledObjects;
+
+namespace Microsoft.CodeAnalysis.CSharp.Symbols
+{
+    internal abstract class RewrittenExtensionMethodSymbol : RewrittenMethodSymbol
+    {
+        protected RewrittenExtensionMethodSymbol(MethodSymbol originalMethod) : base(originalMethod, TypeMap.Empty)
+        {
+            Debug.Assert(originalMethod.ContainingType.GetExtendedTypeNoUseSiteDiagnostics(null) is not null);
+        }
+
+        public sealed override bool IsExtensionMethod => false;
+        public sealed override bool IsVirtual => false;
```

Comment:
> Taking this property as an example, would it be worthwhile to assert that the original method was also not virtual?

I prefer to not do that because we do not control what comes from metadata and I do not want to worry how erroneous modifiers are are handled on source. The current implementation reflects how we decided to see extension methods and I don't really care what the original method says.